### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by meehanti23**
Added AWS and image handling on salad post request

### CODE REVIEW

File: client/src/components/SaladForm.js [hunk 1]
```diff
@@ -2,8 +2,13 @@ import React, { useState } from "react";
...
+import Dropzone from "react-dropzone"
 
 const SaladForm = (props) => {
+    const [images, setImages] = useState([])
+    const [newImageFormData, setNewImageFormData] = useState({
+        image: {}
+    })
 
     const [errors, setErrors] = useState([])
     const [shouldRedirect, setShouldRedirect] = useState(false)
```
Comment (bobbydigital1986) [inline client/src/components/SaladForm.js:14]: It looks like shouldRedirect leverages a newSaladId - it should be set in defaults as well

File: client/src/components/SaladForm.js [hunk 2]
```diff
@@ -80,8 +100,21 @@ const SaladForm = (props) => {
...
+                                    <input {...getInputProps()} />
+                                    <p>Upload Your Salad Image - drag 'n' drop or click to upload</p>
+                                </div>
+                            </section>
+                        )}
+                </Dropzone>
+                </div>
+                <input className="button" type="submit" value="Add" />
+            </form>
+
```
Comment (bobbydigital1986) [inline client/src/components/SaladForm.js]: Extra space

File: server/package.json [hunk 3]
```diff
@@ -55,9 +56,11 @@
     "express-handlebars": "^5.2.0",
     "knex": "^2.4.2",
     "morgan": "^1.10.0",
+    "multer": "^1.4.2",
+    "multer-s3": "^2.9.0",
     "objection": "^3.0.1",
     "objection-unique": "^1.2.2",
-    "passport": "^0.5.0",
+    "passport": "^0.6.0",
```
Comment (bobbydigital1986) [inline server/package.json]: delete change?

File: yarn.lock [hunk 4]
```diff
@@ -4,27 +4,27 @@
 
 "@ampproject/remapping@^2.2.0":
   version "2.2.1"
-  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.2.1.tgz#99e8e11851128b8702cd57c33684f1d0f260b630"
+  resolved "https://registry.npmjs.org/@ampproject/remapping/-/remapping-2.2.1.tgz"
```
Comment (bobbydigital1986) [inline yarn.lock]: @dovakeen118 npm hijacked tim's PR and we're not sure how to resolve :(

**Review by bobbydigital1986 — CHANGES_REQUESTED** [review]

**Review by nwalberts — APPROVED** [review]
looks good!

**Review by bobbydigital1986 — APPROVED** [review]

**Pull request merged**